### PR TITLE
Add verilator/4.034 recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **verilator/4.034**

- [x] Requires https://github.com/conan-io/hooks/pull/200

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by madebr** [comment]
@uilianries This build folder and build module was added in `package_info`.

File: recipes/verilator/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+        try:
+            os.unlink(os.path.join(self._source_subfolder, "src", "config_build.h"))
+        except FileNotFoundError:
+            pass
+
+        if self.settings.compiler == "Visual Studio":
+            tools.replace_in_file(os.path.join(self._source_subfolder, "src", "Makefile_obj.in"),
+                                  "${LINK}", "${PROGLINK}")
+
+        tools.save("msvc_link.sh", textwrap.dedent("""\
```
Comment (uilianries) [inline recipes/verilator/all/conanfile.py:104]: it could be a separated file

Comment (madebr) [reply recipes/verilator/all/conanfile.py]: I moved it to a patch.

Can you please take a look at the remaining blocker concerning builddirs and build_modules?

Comment (madebr) [reply recipes/verilator/all/conanfile.py]: Scrap that, I will push a fix to the hooks repo.

Comment (madebr) [reply recipes/verilator/all/conanfile.py]: @uilianries 
See https://github.com/conan-io/hooks/pull/200

Comment (madebr) [reply recipes/verilator/all/conanfile.py]: The tests are failing due to a recent conan addition...

Comment (uilianries) [reply recipes/verilator/all/conanfile.py]: @madebr thanks for providing that fix!

File: recipes/verilator/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+            autotools = self._configure_autotools()
+            autotools.install(args=self._make_args)
+
+        tools.rmdir(os.path.join(self.package_folder, "bin", "share", "man"))
+        tools.rmdir(os.path.join(self.package_folder, "bin", "share", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "bin", "share", "verilator", "examples"))
+
+        os.unlink(os.path.join(self.package_folder, "bin", "share", "verilator", "verilator-config-version.cmake"))
+        os.rename(os.path.join(self.package_folder, "bin", "share", "verilator", "verilator-config.cmake"),
+                  os.path.join(self.package_folder, "bin", "share", "verilator", "verilator-tools.cmake"))
```
Comment (uilianries) [inline recipes/verilator/all/conanfile.py:122]: Is there a macro or something like that in those cmake files?

Comment (madebr) [reply recipes/verilator/all/conanfile.py:122]: Yes indeed. It is tested in the test_package.
Verilator is a installer package. the script will look for a suitable verilator executable and use it to generate sources.

Comment (uilianries) [reply recipes/verilator/all/conanfile.py:122]: I see, thanks for the explanation

File: recipes/verilator/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+        for fn in glob.glob(os.path.join(self.package_folder, "bin", "share", "verilator", "bin", "*")):
+            print(fn, "->", "..")
+            os.rename(fn, os.path.join(self.package_folder, "bin", os.path.basename(fn)))
+        tools.rmdir(os.path.join(self.package_folder, "bin", "share", "verilator", "bin"))
+
+    def package_id(self):
+        del self.info.settings.compiler
+
+    def package_info(self):
```
Comment (madebr) [inline recipes/verilator/all/conanfile.py:139]:
```suggestion
        tools.rmdir(os.path.join(self.package_folder, "bin", "share", "verilator", "bin"))

    def package_info(self):
```

**Pull request merged**